### DISCUSSION TITLE
Support blog?

### DISCUSSION THREAD

**Discussion by rmannibucau** [dotnet/docfx#10031] [Ideas]
Hi,

Coming from Java land and more particular [Yupiik minisite](https://www.yupiik.io/tools-maven-plugin/mojos.html#minisite) or [jbake](https://jbake.org/docs/2.6.7/) for the static documentation, I'm missing blog option.

It consists in a few features "easy" to do from docfx code standpoint but hard to do from outside on a regular basis (I mean it is doable for a single project but it is not desired for all projects).

In terms of features it is quite simple:

* Enable to add metadata to pages: one or multiple authors, a publication date, one or multiple categories
* Use the metadata to NOT render the not yet published posts in release mode (for docfx it would be publish all in coming watch command for ex and use current system date to filter in build/default command for ex)
* Use the metadata to generate a paginated page for all posts
* Use the metadata to generate a paginated page for all posts *by author*
* Use the metadata to generate a paginated page for all posts *by category*
* (template) Inject at the bottom of every post of the blog section links to the author and category index, a link to all post index page and a next/previous link to browse posts between them (based on the date, potentially the author/category if desired - this is less important while the linking is done to encourage reader to stay)
* (template) Inject reading time and author/category (often next the post title)
* enable to use an icon per category/author (potentially gravatar?)

Note: you can check an example there https://rmannibucau.metawerx.net/simple-bash-cli-for-automotion.html

Due to docfx structure I don't think it is crazy to add blogging support as a built-in feature and it would be great for projects to be able to either just blog or more commonly announce news about the project (releases, new features, EOL, ...).

Hope it makes sense

**Comment by VaclavElias** [comment]
I can just agree. We at https://github.com/stride3d have been using **docfx** for API Docs and [11ty](https://www.11ty.dev/docs/) for the website and blog. I was thinking to use only one static site generator just to have less gears to maintain in our open source project. 

If PRs would be open for such functionality, I would suggest to get inspired by 11ty to bring some features/functionality which would help adding also blog posts. 

If we could generate this blog and post pages https://www.stride3d.net/blog/ with docfx, then we would use just docfx across our websites.

**Comment by rmannibucau** [comment]
If it helps I did a custom solution which covers that thanks templating (theme in docfx) at https://github.com/rmannibucau/SBSharp and here is a sample https://github.com/dotnetbirdie/dotnetbirdie.github.io/tree/main . The trick is really to delegate the decision to the template/theme giving enough information/context to avoid to have to code anything but it is very doable.
